### PR TITLE
Gui

### PR DESCRIPTION
**Pull request by lima-limon-inc**
Features implemented:

- All the gui components
- P for pausing
- Clicking on a cell changes its state

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,18 +1,161 @@
...
     }
 }
+
+fn check_for_events(keys: &[KeyCode; 10], grid: &mut Grid, pause: bool) -> bool {
+    match keys {
+        [KeyCode::Up, KeyCode::Up, KeyCode::Down, KeyCode::Down, KeyCode::Left, KeyCode::Right, KeyCode::Left, KeyCode::Right, KeyCode::A, KeyCode::B] =>
+        {
+            grid.kill_all();
+
+            // Hmm, I wonder what this does 🤔
```
Comment (dsocolobsky) [inline src/main.rs]: This is quite confusing, not doing anything for me, and Konami code ends in `BA` not `AB`

File: src/main.rs [hunk 2]
```diff
@@ -1,18 +1,161 @@
...
+
     loop {
-        clear_background(RED);
+        clear_background(GRAY);
+        if let Some(key) = get_last_key_pressed() {
+            keys[offset] = key;
+            offset += 1;
+            if offset == keys.len() {
+                offset = 0;
+            }
```
Comment (dsocolobsky) [inline src/main.rs:50]: What does this do?

**Review by dsocolobsky — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 1 above)
Comment (lima-limon-inc) [reply src/main.rs]: Changed the sequence in latest commit

File: src/main.rs [hunk 3]
```diff
@@ -1,18 +1,161 @@
...
-        draw_circle(screen_width() - 30.0, screen_height() - 30.0, 15.0, YELLOW);
+                draw_rectangle(offset_x, offset_y, cube_width, cube_height, color);
+            }
+        }
 
-        draw_text("IT WORKS!", 20.0, 20.0, 30.0, DARKGRAY);
+        if is_mouse_button_pressed(MouseButton::Left) {
+            let (mouse_x, mouse_y) = mouse_position();
+            let mouse_x = (mouse_x / cube_width).ceil() - 1.0;
+            let mouse_y = (mouse_y / cube_height).ceil() - 1.0;
```
Comment (LeanSerra) [inline src/main.rs]: Maybe change this to `floor` instead of ceil so you don't have to subtract 1

**Review by LeanSerra — APPROVED** [review]
LGTM

File: src/main.rs (hunk 3 above)
Comment (lima-limon-inc) [reply src/main.rs]: Changed here: https://github.com/lima-limon-inc/Conways-game-of-life/pull/3/commits/01e1d3e7b11d326fd7ca7a9886435c2282ae9a57

Thank you Lean ^_^

File: src/main.rs (hunk 2 above)
Comment (lima-limon-inc) [reply src/main.rs:50]: It's a secret ;)

(It's part of an easter egg 🐰 🥚 )

**Pull request merged**